pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -184,7 +184,7 @@ func BuildDefaultAPIServer(options configapi.MasterConfig) (*apiserveroptions.Se
 		master.DefaultAPIResourceConfigSource(),
 	)*/
 	// the order here is important, it defines which version will be used for storage
-	storageFactory.AddCohabitatingResources(extensions.Resource("jobs"), batch.Resource("jobs"))
+	storageFactory.AddCohabitatingResources(batch.Resource("jobs"), extensions.Resource("jobs"))
 	storageFactory.AddCohabitatingResources(extensions.Resource("horizontalpodautoscalers"), autoscaling.Resource("horizontalpodautoscalers"))
```

Comment:
@DirectXMan12 should the same storage version switch be done for HPA? I assume we'll want a similar test (store as extensions/v1beta1 HPA, read/write and ensure the update was stored as autoscaling/v1 HPA)